### PR TITLE
Support disabling animations in the CLI

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
This is just the first step; once this lands we need to also look at reducing animation in subinvocations of `pub` and using a less verbose and more static reporter in `flutter test`. The latter would help with CI logs, too.

File: packages/flutter_tools/lib/src/flutter_features.dart [hunk 1]
```diff
@@ -47,6 +47,14 @@ class FlutterFeatureFlags implements FeatureFlags {
   @override
   bool get isFlutterWebWasmEnabled => isEnabled(flutterWebWasm);
 
+  @override
+  bool get isCliAnimationEnabled {
+    if (_platform.environment['TERM'] == 'dumb') {
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/flutter_features.dart:52]: never heard of this setting. I wonder, does anything modern still use this, or will this only affect Flutter developers on an old 70s teletype?

**Review by christopherfujino — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/flutter_features.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter_tools/lib/src/flutter_features.dart:52]: I'm not sure, but from what I can tell it's what some software uses to indicate that they don't want any fancy shenanigans happening on the terminal, so, might as well, right...

**Pull request merged**